InvenTree/part/models.py

```diff
@@ -1465,16 +1465,16 @@ def supplier_count(self):
         return self.supplier_parts.count()
 
     @property
-    def has_pricing_info(self):
+    def has_pricing_info(self,internal=False):
         """ Return true if there is pricing information for this part """
-        return self.get_price_range() is not None
+        return self.get_price_range(internal=internal) is not None
```

Comment:
Should we have a fallback to BOM pricing if there is no internal price?

i.e.

```python
if internal_price is not None:
    return internal_price
elif bom_price is not None:
    return bom_price
else:
    return None
```


@matmair - you have more direct knowledge about the internal pricing structure, do you want to weigh in here?

Reply:
This change will only be effective when PART_BOM_USE_INTERNAL_PRICE is True. 
I was thinking that it might be sensible to use internal pricing when a part has an internal price (and configured to be used) and if no internal pricing is set then use the supplier pricing. Should the supplier not have pricing then the Error should still be displayed.  Your thoughts on this?

Reply:
@rgilham - agreed. Is this what your current implementation does?

Reply:
@SchrodingersGat that check is already in place
https://github.com/inventree/InvenTree/blob/be85de4a7b5dc6d328fe1d519dd592fdba76b76e/InvenTree/part/models.py#L1584-L1587